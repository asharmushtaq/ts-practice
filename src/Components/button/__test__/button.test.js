import React from 'react';
import Button from '../button';
import ReactDOM from 'react-dom'
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect'
import renderer from 'react-test-renderer'

it("from react itself", () => {
    const div = document.createElement('div')
    ReactDOM.render(<Button />, div)
})

it("from external render", () => {
    const { getByTestId } = render(<Button label='hello' />)
    expect(getByTestId('button')).toHaveTextContent('hello')
})

it('snapshot testing', () => {
    const tree = renderer.create(<Button label={'changed'} />).toJSON()
    expect(tree).toMatchSnapshot()
})
