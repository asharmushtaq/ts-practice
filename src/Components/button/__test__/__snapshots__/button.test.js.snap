// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing 1`] = `
<div
  className="button-style"
  data-testid="button"
>
  you 
  changed
</div>
`;
